code/datums/wounds/robotic/robotic_blunt.dm

```diff
@@ -0,0 +1,1027 @@
+				chance_mult *= 1.5
+
+		if (attacking_atom.loc != victim)
+			chance_mult *= 3 // encourages people to get other people to beat the shit out of their limbs
+	if (prob(percussive_maintenance_repair_chance * chance_mult))
+		handle_percussive_maintenance_success(attacking_item)
+	else
+		handle_percussive_maintenance_failure(attacking_item)
+
+/// Called when percussive maintenance succeeds at its random roll.
+/datum/wound/blunt/robotic/proc/handle_percussive_maintenance_success(attacking_item)
+	victim.visible_message(span_green("[victim]'s [limb.plaintext_zone] rattles from the impact, but looks a lot more secure!"), \
+		span_green("Your [limb.plaintext_zone] rattles into place!"))
+	remove_wound()
+
+/// Called when percussive maintenance faisl at its random roll.
+/datum/wound/blunt/robotic/proc/handle_percussive_maintenance_failure(attacking_item)
+	to_chat(victim, span_warning("Your [limb.plaintext_zone] rattles around, but you don't sense any sign of improvement."))
+
+/datum/wound/blunt/robotic/proc/victim_moved(datum/source, atom/old_loc, dir, forced, list/old_locs)
+	SIGNAL_HANDLER
+
+	var/overall_mult = 1
+
+	var/obj/item/stack/gauze = limb.current_gauze
+	if (gauze)
+		overall_mult *= gauze.splint_factor
+	if (!victim.has_gravity(get_turf(victim)))
+		overall_mult *= 0.5
+	else if (victim.body_position == LYING_DOWN || (!forced && victim.move_intent == MOVE_INTENT_WALK))
+		overall_mult *= 0.25
+	if (victim.has_status_effect(/datum/status_effect/determined))
+		overall_mult *= ROBOTIC_WOUND_DETERMINATION_MOVEMENT_EFFECT_MOD
+
+	overall_mult *= get_buckled_movement_consequence_mult(victim.buckled)
+
+	if (can_daze())
+		var/shake_chance = head_movement_shake_chance
+		var/dizzy_chance = head_movement_dizzy_chance
+
+		shake_chance *= overall_mult
+		dizzy_chance *= overall_mult
+		var/daze_mult = LERP(1, 1.2, rand())
+
+		var/duration = (daze_mult * head_movement_base_shake_duration)
+		var/strength = (daze_mult * head_movement_base_shake_intensity)
+
+		if (can_do_movement_shake && can_shake_camera(duration, strength * head_movement_shake_dizziness_overtake_mult) && prob(shake_chance))
+			shake_camera(victim, duration = duration, strength = strength)
+
+		if (prob(dizzy_chance))
+			victim.adjust_dizzy_up_to(head_movement_dizzy_base_duration * daze_mult, daze_dizziness_maximum_duration)
+
+	if (limb.body_zone == BODY_ZONE_CHEST)
+		if (prob(chest_movement_nausea_chance * overall_mult))
+			shake_organs_for_nausea(chest_movement_base_nausea_score, max_nausea_duration)
+
+		if (prob(chest_movement_organ_damage_chance * overall_mult))
+			attack_random_organs(get_chest_movement_organ_damage(), chest_movement_organ_damage_individual_max)
+
+/datum/wound/blunt/robotic/proc/can_shake_camera(strength, duration)
+	var/datum/status_effect/dizziness/dizzy_effect = get_dizzy_effect()
+	if (isnull(dizzy_effect) || !dizzy_effect.applying_dizziness)
+		return TRUE
+	var/amount = (dizzy_effect.duration - world.time) / 10
+	if (amount <= 0)
```

Comment:
Don't think so, but I mainly just copied over bits and pieces from the proc to be more 1:1, it could be sanity checking

Again, I can remove this if youd like